src/components/Modal/Modal.tsx

```diff
@@ -1,27 +1,43 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { ModalOverlay } from "./ModalOverlay";
 
 export interface ModalProps {
   children: React.ReactNode;
   name: string;
-  isOpen: boolean; // Added prop
-  onClose?: () => void; // Added prop
+  isOpen: boolean;
+  onClose?: () => void;
 }
 
 export const Modal = ({ children, name, isOpen, onClose }: ModalProps) => {
-  // Early return if the modal isn't open
-  if (!isOpen) {
-    return null;
+  const [isVisible, setIsVisible] = useState(isOpen); // Track when the modal should be visible
+
+  useEffect(() => {
+    if (isOpen) {
+      // When isOpen changes to true, make the modal visible
+      setIsVisible(true);
+    } else {
+      // When isOpen changes to false, wait for the closing animation, then hide the modal
+      const timer = setTimeout(() => {
```

Comment:
I wouldn't worry about a timer here. We can simply `setIsVisible(false)` without worrying about a timer.

Reply:
Address comment, removed timer, updated animation, updated state variables to reflect changes